README.md

```diff
@@ -1,60 +1,349 @@
+    # "${self}/hosts/shared/optional/containers.nix"
+    # "${self}/modules/nixos/custom-module.nix"
+  ];
+
+  # Host-specific configuration
+  host = {
+    device.isHeadless = false; # Set to true for servers
+  };
+
+  # Add your system configuration here
+  # networking.hostName is automatically set from directory name
+}
+```
+
+Create `hosts/{device-type}/{hostname}/hardware.nix`:
+```nix
+{ inputs, ... }:
+{
+  imports = [
+    # Include relevant hardware modules
+    inputs.nixos-hardware.nixosModules.common-cpu-amd
+    inputs.nixos-hardware.nixosModules.common-pc-ssd
+    
+    # For laptops, also include:
+    # inputs.nixos-hardware.nixosModules.common-pc-laptop
+  ];
+
+  # Boot configuration
+  boot.loader = {
+    systemd-boot.enable = true;
+    efi.canTouchEfiVariables = true;
+  };
+
+  # Filesystem configuration (use disko for declarative disk setup)
+  fileSystems."/" = {
+    device = "/dev/disk/by-label/nixos";
+    fsType = "ext4";
+  };
+
+  # Add hardware-specific configuration
+}
+```
+
+### 3. Add Hardware Acceleration (Optional)
+
+If your host supports CUDA or ROCm, add it to the acceleration lists in `flake.nix`:
+```nix
+accelerationHosts = {
+  cuda = [
+    "nixmi"
+    "winix"
+    "your-new-host"  # Add here for CUDA support
+  ];
+  rocm = [
+    "your-amd-host"  # Add here for ROCm support
+  ];
+};
+```
+
+### 4. Build and Test
+```bash
+# Build the configuration (don't switch yet)
+sudo nixos-rebuild build --flake .#your-new-host
+
+# Test the configuration 
+sudo nixos-rebuild test --flake .#your-new-host
+
+# Switch to the new configuration
+sudo nixos-rebuild switch --flake .#your-new-host
+```
+
+## Creating New Users
+
+To add a new user configuration:
+
+### 1. Create User Directory
+```bash
+mkdir -p home/newuser
 ```
 
+### 2. Create User Configuration Files
+
+Create host-specific configurations in `home/newuser/{hostname}.nix`:
+```nix
+{ pkgs, lib, ... }:
+{
+  imports = [
+    # Import shared configurations
+    ./features/cli              # Common CLI tools
+    ./features/desktop/common   # Desktop environment basics
+  ];
+
+  # User-specific configuration
+  home = {
+    username = "newuser";
+    homeDirectory = "/home/newuser";
+    stateVersion = "25.05";
+  };
+
+  # Add user-specific packages and configuration
+  programs = {
+    git = {
+      userName = "Your Name";
+      userEmail = "your.email@domain.com";
+    };
+  };
+}
+```
+
+Create feature modules in `home/newuser/features/`:
+```bash
+mkdir -p home/newuser/features/{cli,desktop,development}
+```
+
+### 3. Link User to Hosts
+
+The auto-discovery system will automatically link users to hosts if:
+- A file `home/{username}/{hostname}.nix` exists
+- The hostname matches an existing host configuration
+
+### 4. Test User Configuration
+```bash
+# Build home-manager configuration
+home-manager build --flake .#newuser@hostname
+
+# Switch to new configuration  
+home-manager switch --flake .#newuser@hostname
+```
```

Comment:
These sections should be extracted to their own dedicated files that are linked to